### ISSUE TITLE
spurious fallback call to render function when getStaticPaths() returns empty paths array

### ISSUE THREAD

**Issue by godmar** [vercel/next.js#17153]
# Bug report

## Describe the bug

I'm trying to use the getStaticPaths, returning `{ paths: [], fallback: true }` since I have not yet decided which paths I want to prerender.

When using the development server or when running `next build`, my component is called server-side with an empty props `{}` and a router object that looks like this:
```
ServerRouter {
  route: '/item/[id]',
  pathname: '/item/[id]',
  query: {},
  asPath: '/item/[id]',
  basePath: '',
  events: undefined,
  isFallback: true
}
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

I used the [`create-next-app`](https://nextjs.org/learn/basics/create-nextjs-app/setup) and added 1 file `/pages/item/[id].js` like so:
```
import React from 'react';
import { useRouter } from 'next/router';

export default function ProcessorPage(props) {
  const router = useRouter();
  console.log(`Rendering ProcessorPage, props are`);
  console.log(props);
  console.log(`Rendering ProcessorPage, router is`);
  console.log(router);
  return (<div>Current props are: {JSON.stringify(props)} and router is {JSON.stringify(router)} </div>)
}

export async function getStaticPaths() {
  const rc = {
    //paths: [{ params: { id : "1" }}],
    paths: [],
    fallback: true
  };
  return rc;
}

export async function getStaticProps(context) {
  return {
      props: context
  };
}
```

## Expected behavior

I'm not really sure what to expect. I thought I could write the component to simply rely on what's in the `params` property. Instead, it appears as though I'll have to write code to ignore the spurious fallback render request?  Or perhaps the request is not spurious and I'm supposed to render something here. I'm confused.

If I provide at least one element in `paths`, such as `{ params: { id : "1" }}` I do not see the spurious call.

## System information

- OS: Linux
- Version of Next.js: 9.5.3
- Version of Node.js: 12.16.1

**Comment by ijjk** [comment]
Hi, this is the expected behavior for `fallback: true` mode. You can handle this loading state by using the `isFallback` field provided on the router. See here for more information on this in our docs https://nextjs.org/docs/basic-features/data-fetching#fallback-true

**Comment by godmar** [comment]
Thank you. In my application, the component actually fetches some data from a backend, then renders it. We want to `next export` this to be completely static (no JS involved server side during deployment, all served with nginx).  I tried this, and it fails with "Pages with `fallback` enabled in `getStaticPaths` can not be exported. See more info here: https://err.sh/next.js/ssg-fallback-true-export"

I'm now starting to think that `getStaticPath` is not for my use case, in which I would like an exported page to be populated by an API server side for a subset of routes for SEO purposes, and the fallback to be client-initiated API accesses. I suppose in order to get the functionality of `getStaticPaths` I will need an export map. 

When I try to create an export map, I do:
```
     return {
          '/item/1': { page: '/item/[id]', query: { id: '1' } },
        };
```
but this fails with:
"you provided query values for /item/1 which is an auto-exported page. These can not be applied since the page can no longer be re-rendered on the server. To disable auto-export for this page add `getInitialProps`"

Omitting the `query` parameter builds and exports a page, but it doesn't have the information for item `1` in it. Also, it's not clear how to do the fallback here: if I've exported `/item/1`, `/item/2`, will `/item/3` still work even though it's not in the export map?

The other proposed option is `getInitialProps`, but that's [deprecated in favor of getServerSideProps or getStaticProps](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps). An attempt to use `getServerSideProps` fails with "pages with `getServerSideProps` can not be exported. See more info here: https://err.sh/next.js/gssp-export". When trying `getStaticProps` it alerts me that getStaticPaths is then required and the circle closes with where I've started.

Perhaps I'm trying to be too magical here with my use case - which is, again:

- nginx for static assets
- no JavaScript SSR involved in requests
- API for data (can be accessed both during export and from the client)
- SEO optimization for subset of `/item/`'s by exporting them to static HTML (say `/item/1`, `/item/2`)
- traditional client-initiated API requests for other `/item/` via data API.

I'm sorry this has developed into a how-to question. I generally really appreciate how much effort you put into documentation and the error pages you link to are usually very helpful as well. I've read [Next-gen Static Site Generation (SSG) Support](https://nextjs.org/blog/next-9-3#next-gen-static-site-generation-ssg-support) but am still confused as to how this relates to `export`
